invenio/modules/access/upgrades/access_2015_05_06_accROLE_accACTION_accARGUMENT_id.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Upgrade recipe."""
+
+import warnings
+
+from invenio.ext.sqlalchemy import db
+from invenio.legacy.dbquery import run_sql
+from invenio.modules.upgrader.api import op
+
+from sqlalchemy.exc import OperationalError
+
+depends_on = ['invenio_release_1_2_0']
+
+
+def info():
+    """Info."""
+    return "Add a autoincrement id in accROLE_accACTION_accARGUMENT table"""
+
+
+def do_upgrade():
+    """Implement your upgrades here."""
+    # table accROLE_accACTION_accARGUMENT
+
+    # add "id" column
+    op.add_column('accROLE_accACTION_accARGUMENT',
+                  db.Column('id', db.Integer(15, unsigned=True),
+                            nullable=False))
+    # set all ids
+    records = run_sql(
+        """SELECT id_accROLE, id_accACTION, id_accARGUMENT, argumentlistid
+           FROM accROLE_accACTION_accARGUMENT AS aaa
+           ORDER BY aaa.id_accROLE, aaa.id_accACTION,
+                    aaa.id_accARGUMENT, aaa.argumentlistid """)
+    for index, rec in enumerate(records):
+        run_sql("""UPDATE accROLE_accACTION_accARGUMENT
+                   SET id = %s
+                   WHERE id_accROLE = %s AND
+                         id_accACTION = %s AND
+                         id_accARGUMENT = %s AND
+                         argumentlistid = %s """,
+                (index + 1, rec[0], rec[1], rec[2], rec[3]))
+    # remove primary key
+    try:
+        op.drop_constraint(None, 'accROLE_accACTION_accARGUMENT',
```

Comment:
if you do:
```python
op.drop_constraint(None, 'accROLE_accACTION_accARGUMENT',type_='primary')
op.add_column('accROLE_accACTION_accARGUMENT', db.Column('id', db.Integer(15, unsigned=True), nullable=False, autoincrement=True))
```
Mysql create a id column by initialized as zero.

p.s. If I try (after drop the constraint) to create the id column also as primary key:

```python
op.add_column('accROLE_accACTION_accARGUMENT', db.Column('id', db.Integer(15, unsigned=True), nullable=False, autoincrement=True, primary_key=True))
```

Mysql returns error:
```
OperationalError: (_mysql_exceptions.OperationalError) (1075, 'Incorrect table definition; there can be only one auto column and it must be defined as a key') [SQL: u'ALTER TABLE `accROLE_accACTION_accARGUMENT` ADD COLUMN id INTEGER(15) UNSIGNED NOT NULL AUTO_INCREMENT']
```